### PR TITLE
Choose Pieces and Choose Spaces

### PR DESCRIPTION
**Pull request by alexanderGGreenberg**
It lets them choose which pieces they want to place, and decides if a space on the board is a valid space.

### CODE REVIEW

File: src/shipBoard.java [hunk 1]
```diff
@@ -1,10 +1,91 @@
...
-		// TODO Auto-generated method stub
+		reader = new Scanner(System.in);
+		/*
+		 * Part of the game where they set up the board
+		 * Cycles through all the remaining pieces until user decides to place one
+		 * Once chosen keeps prompting for location until a correct one is given
+		 * Will need to let them choose direction, not let repeats
+		 */
+		
+		ArrayList<String> gamePieces = new ArrayList(5);
```
Comment (alexanderGGreenberg) [inline src/shipBoard.java]: This is an ArrayList, which is really just like array, except in is an object, so it has useful methods like .add(thing) and .remove(thing or index). 

In an arrayList, we want to specify that each item is a String, so we can treat each item like a String(as opposed to any other object).  So we say <String>

File: src/shipBoard.java [hunk 2]
```diff
@@ -1,10 +1,91 @@
...
+				/*First lets make sure the last character is valid*/
+				validMoveBool = validMove(move);
+				
+			}
+		}
+	}
+	
+	public static boolean validMove(String input) {
+		int finalCharToInt;
+		try {
```
Comment (alexanderGGreenberg) [inline src/shipBoard.java]: The point of the try catch here is that if an error is thrown when we try to convert the String to integer or vice versa, that means it wasn't a proper input, so we catch the error and return false.

File: src/shipBoard.java [hunk 3]
```diff
@@ -1,10 +1,91 @@
...
+		
+		ArrayList<String> gamePieces = new ArrayList(5);
+		gamePieces.add("Aircraft Carrier");
+		gamePieces.add("Battleship");
+		gamePieces.add("Submarine");
+		gamePieces.add("Crussier");
+		gamePieces.add("Destroyer");
+		while(!gamePieces.isEmpty()) {
+			Iterator<String> it =  gamePieces.iterator();
+			while(it.hasNext()) {
```
Comment (alexanderGGreenberg) [inline src/shipBoard.java]: Iterators are weird, but they let us delete objects in the ArrayList without having to worry about accidentally referencing a deleted object.  The iterator cycles through the list until there are no more elements left, it.next() the element the iterator is currently processing.  it.remove() removes the element it is processing.

File: src/shipBoard.java [hunk 4]
```diff
@@ -1,10 +1,91 @@
...
+				System.out.println(it.next());
+				String response = reader.nextLine();
+				if(!response.isEmpty()){
```
Comment (alexanderGGreenberg) [inline src/shipBoard.java]: They just hit enter, meaning they want to skip this piece and move onto the next.

**Comment by alexanderGGreenberg** [comment]
Give this a look, I added a comment on most of the Java stuff that doesn't pop up much in other languages.  Leave a comment on anything that you think needs changing or just doesn't make sense :).

File: src/shipBoard.java [hunk 5]
```diff
@@ -1,10 +1,91 @@
...
+				validMoveBool = validMove(move);
+				
+			}
+		}
+	}
+	
+	public static boolean validMove(String input) {
+		int finalCharToInt;
+		try {
+			finalCharToInt = input.charAt(input.length() - 1)-'a'+1;
```
Comment (alexanderGGreenberg) [inline src/shipBoard.java]: this is a funky way of making it so we don't have have a different case for 'a', 'b' 'c' etc...  java makes char to int conversation really simple, and by doing this a = 1, b = 2...  This way a simple if statement works.  Confession: stole the conversion off of stack overflow but its pretty simple and makes sense so its good to remember.

File: src/shipBoard.java [hunk 6]
```diff
@@ -1,10 +1,91 @@
+import java.awt.List;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.Iterator;
+
 
 public class shipBoard {
+	public static boolean gameOver = false;
+	private static Scanner reader;
```
Comment (alexanderGGreenberg) [inline src/shipBoard.java]: Just how we take input from the console in Java, you can use this to get the next integer, string or whole line they enter.

File: src/shipBoard.java [hunk 7]
```diff
@@ -1,10 +1,91 @@
...
+		}
+		
+		//part of the game where guess locations
 		System.out.println("hello battleship!!!!!!!!");
 		System.out.println("Hey Alex");
+		System.out.println("Welcome to battleship");
+		
+		while(!gameOver) {
+			System.out.println("next move");
+			for(int i = 0; i <=9; i++) {
```
Comment (alexanderGGreenberg) [inline src/shipBoard.java]: print out game board between each move, ?'s represent the spaces where they haven't guessed yet

File: src/shipBoard.java [hunk 8]
```diff
@@ -1,10 +1,91 @@
...
+		try {
+			finalCharToInt = input.charAt(input.length() - 1)-'a'+1;
+		} catch(Exception e) {
+			return false;
+		}
+		if( finalCharToInt > 0 && finalCharToInt <= 10) {
+			/*Next lets make sure that the first number is valid*/
+			int firstNum;
+			try {
+				firstNum = Integer.parseInt(input.substring(0, input.length() - 1));
```
Comment (alexanderGGreenberg) [inline src/shipBoard.java]: take the substring that goes from character 0 and ends before the last character.  This is pretty much always how you split up a string in java.

**Comment by youngpines** [comment]
gr8 changes

**Pull request merged**